### PR TITLE
Reformat dev_clone_array in device.py

### PR DESCRIPTION
**Pull request by Unevilicorn**
#1409

### CODE REVIEW

**Comment by simonetgordon** [comment]
Hello,

Thanks for the contribution!

There are a few things you need to address:

1) The docstring examples are not exhaustive. Please follow the [Contributer guide](https://lets-unify.ai/ivy/deep_dive/11_docstring_examples.html#docstring-examples) to add all necessary examples.

2) There are formatting issues being picked up by the lint check. You can see what needs to be fixed under the 'details' of the lint check.

Kind Regards, 

Simone

**Comment by Unevilicorn** [comment]
Hi,

I have added more exhaustive examples.
However I'm unsure about the linting, as previously it said illegal character in one of the lines, when I looked at it, it seemed fine, also it mentioned line 94 on setup.py was too long, but when I checked the line on that commit, it only had a closing bracket. Am I missing something?

Thanks

**Comment by simonetgordon** [comment]
Hi,

Thanks for making the changes. If you could just resolve the conflicts with the master branch so that only `ivy/functional/ivy/device.py` is changed, that'd be great 😊 

Kind Regards, 

Simone

**Comment by Unevilicorn** [comment]
Hi,

I have reverted the changes and remerged, however I can't pass the linting test without changing some other files, since the linting error comes from `test_device.py`.

```
ivy_tests/test_ivy/test_functional/test_core/test_device.py:442:89: E501 line too long (92 > 88 characters)
ivy_tests/test_ivy/test_functional/test_core/test_device.py:483:89: E501 line too long (92 > 88 characters)
ivy_tests/test_ivy/test_functional/test_core/test_elementwise.py:70:5: E265 block comment should start with '# '
```

Thanks

**Comment by simonetgordon** [comment]
Hi,

Yes you don't need to change those other files, and thanks for resolving the conflicts.

It seems you are missing an 'out' argument in the function signature, as per the tan() example in the contributer guide. You will then need to write examples using this argument (point 4 in the list).

Thanks!

Simone

**Comment by Unevilicorn** [comment]
Hi,

There wasn't an out argument before, I have now added it and included the relevant examples.

Thanks

File: ivy/functional/ivy/device.py [hunk 1]
```diff
@@ -1087,10 +1090,73 @@ def dev_clone_array(x, devices):
...
+    >>> y = ivy.DevClonedItem({})
+    >>> print(y)
+    DevClonedItem({})
+    >>> ivy.dev_clone(x, ['cpu'], out=y)
+    >>> print(y)
+    DevClonedItem({'cpu': array([1, 2])})
+
+    >>> x = ivy.Container({'a': ivy.native_array([3, 4])})
+    >>> y = ivy.DevClonedItem({})
+    >>> print(y)
```
Comment (simonetgordon) [inline ivy/functional/ivy/device.py:1143]: Don't need to print this out

Comment (Unevilicorn) [reply ivy/functional/ivy/device.py]: Printing is removed now

**Comment by simonetgordon** [comment]
Hi,

Are the container examples and out argument examples running on a script for you? They are not working for me and I think this is causing a warning to be raised in test-docstrings/run-docstring-tests (you can check this out in 'details' under 'Run docstring tests'). 

Take a look and see if you can fix them up. If not, it might be necessary to just take out the 'out' argument again. Also see if you can get the container examples working. The other two sets of examples seem to run fine.

Thanks!

Simone

**Comment by Unevilicorn** [comment]
Hi,

I am able to run container examples, and able to get the docstring warning to go away if I remove the `out` argument.
I think this could be caused by different framework, If I use numpy as backend then it runs fine, but if I switch to pytorch, some error shows up.

Thanks

**Comment by simonetgordon** [comment]
Hi, 

That's very useful to know, thanks! Also great that no warnings are coming up, looks ready to merge. Will just wait for the other tests to run just in case.

Thanks for the contribution 😊

**Comment by Unevilicorn** [comment]
Thank you for all the help!

**Pull request merged**